climada_petals/hazard/tc_tracks_forecast.py

```diff
@@ -188,80 +198,154 @@ def fetch_bufr_ftp(target_dir=None, remote_dir=None):
+            # Location of max wind
+            sig_values = ec.codes_get_array(bufr, "#%d#meteorologicalAttributeSignificance" % rank3)
+            if len(sig_values) == 1:
+                significanceWind = sig_values[0]
+            else:
+                for j in range(len(sig_values)):
+                    if sig_values[j] != ec.CODES_MISSING_LONG:
+                        significanceWind = sig_values[j]
+                        break
+            # max_wind of all ensemble members at ind_timestep
+            if significanceWind == 3:
+                wnd_temp = ec.codes_get_array(bufr, "#%d#windSpeedAt10M" % (ind_timestep + 1))
+            else:
+                raise ValueError('unexpected meteorologicalAttributeSignificance=', significance)
 
-        if fcast_rep is None:
-            fcast_rep = self._find_delayed_replicator(
-                meta_query(bufr, '%unexpanded_descriptors')
-            )
+            # check dimention of the variables, and replace missing value with NaN
+            lat = self._check_variable(lat_temp, ens_no)
+            lon = self._check_variable(lon_temp, ens_no)
+            pre = self._check_variable(pre_temp, ens_no)
+            wnd = self._check_variable(wnd_temp, ens_no)
 
-        # query the bufr message
+            # appending values into dictionaries
+            for ind_ens in range(len(ens_no)):
+                latitude[ind_ens] = np.append(latitude[ind_ens], lat[ind_ens])
+                longitude[ind_ens] = np.append(longitude[ind_ens], lon[ind_ens])
+                pressure[ind_ens] = np.append(pressure[ind_ens], pre[ind_ens])
+                max_wind[ind_ens] = np.append(max_wind[ind_ens], wnd[ind_ens])
+
+
+        # storing information into a dictionary
         msg = {
             # subset forecast data
-            'significance': data_query(bufr, fcast_rep + '> 008005'),
-            'latitude': data_query(bufr, fcast_rep + '> 005002'),
-            'longitude': data_query(bufr, fcast_rep + '> 006002'),
-            'wind_10m': data_query(bufr, fcast_rep + '> 011012'),
-            'pressure': data_query(bufr, fcast_rep + '> 010051'),
-            'timestamp': data_query(bufr, fcast_rep + '> 004024'),
+            'latitude': latitude,
+            'longitude': longitude,
+            'wind_10m': max_wind,
+            'pressure': pressure,
+            'timestamp': timesteps_int,
 
             # subset metadata
-            'wmo_longname': data_query(bufr, '/001027'),
-            'storm_id': data_query(bufr, '/001025'),
-            'ens_type': data_query(bufr, '/001092'),
-            'ens_number': data_query(bufr, '/001091'),
+            'wmo_longname': ec.codes_get(bufr, 'longStormName').strip(),
+            'storm_id': ec.codes_get(bufr, 'stormIdentifier').strip(),
+            'ens_type': ec.codes_get_array(bufr, 'ensembleForecastType'),
+            'ens_number': ec.codes_get_array(bufr, "ensembleMemberNumber"),
```

Comment:
I'm worrying if this could cause some confusion. Let's say we have track data only for ensemble members 2, 12 and 22. Then we've indexed our results dictionary by as `{0: [...], 1: [...], 2: [...]}` but here in the metadata we're recording the ensemble number as `[2, 12, 22]`. Do we want instead to index results as `{2: [...], 12: [...], 22: [...]}`?

Or is this just not worth worrying about?

Reply:
It doesn't matter. The dictionary `{0: [...], 1: [...], 2: [...]}` is only used in this methods to temporarily store the data and later being distribute to the list of xarray dataset (the standard structure of TCTracks. The indices of the dictionary are not returned to the TCTracks class.

Reply:
ahhhh yeah then this didn't matter at all